branch 18 add 検索機能追加

L2206693309 opened the pull request:
受講生の検索機能追加
受講生コース情報にステータスを追加

L2206693309 committed:
29回コミット add 検索機能追加
コントローラーに検索用のメソッド追加
サービスに上記のメソッドから呼ばれる、検索用のメソッドを追加
リポジトリ(StudentRepository.java)に上記のメソッドから呼ばれる、検索用のメソッド追加
リポジトリ(studentRepository.xml)に上記のメソッドから呼ばれる、検索用のSQL文を追加
コントローラー, サービス, リポジトリの自動テストの実装
リポジトリの自動テストのため、schema.sqlとdata.sqlを変更
StudentCourses.javaに、新たに Integer statusId, String statusを追加

KeitaUenishi commented on src/main/java/raisetech/StudentManagement/service/StudentService.java:
```diff
@@ -92,6 +92,80 @@ public List<StudentCourses> studentsCoursesJava() {
+   * @return 受講生詳細一覧
+   */
+  public List<StudentDetail> searchStudents(Students targetStudents, Integer isDeleted) {
+    List<Students> students = repository.findAllStudents();
+    List<StudentCourses> studentCourses = repository.findAllStudentCourseList();
+    List<Students> returnStudents = new ArrayList<>();
```
現在の実装では一度全ての受講生情報を取得し、ロジック側で切り分けていますがrepository側のSQLで検索して特定のデータを取得できるようにした方が実装的には良いですね！
そうするとこのメソッド内のfor文の処理は必要無くなると思います！

L2206693309 replied:
ripositoryのSQL文で指定された条件に合うレコードのみを選択するよう変更します。実装後にDiscordにてお知らせします。

L2206693309 committed:
30回コミット change 検索機能仕様変更
StudentRepository.javaに、検索用メソッドを追加を追加
StudentRepository.javaのsearchStudentCoursesを、引数の許容のため、要求する引数を変更したものを追加
studentRepository.xmlに、検索条件を満たすレコードを選択するSQL文を追加
StudentService.javaのsearchStudentsの仕様変更
上記変更に合わせ、全ての自動テストを変更, 追加
